rosbag2_storage_default_plugins/test/rosbag2_storage_default_plugins/sqlite/storage_test_fixture.hpp

```diff
@@ -70,6 +70,7 @@ class StorageTestFixture : public TemporaryDirectoryFixture
       serialized_data->buffer, serialized_data->buffer_capacity, message);
 
     assert(written_size == message_size - 1);  // terminated null character not counted
```

Comment:
nitpick: replacing this `assert` with `ASSERT_THAT` or similar is probably the better option.

Reply:
Does ASSERT_THAT overcome the unused warning ?

Reply:
Yes, other than `assert` it does not "vanish" in a release mode build.